### DISCUSSION TITLE
HTTPS server on WT32_ETH01?

### DISCUSSION THREAD

**Discussion by PyCoder040** [khoih-prog/WebServer_WT32_ETH01#11] [General]
Will this library ever support an HTTPS server on the WT32_ETH01?

I also notice you have a library called EthernetWebServer_SSL, but it seems to support all Ethernet chips except LAN8720 which is part of the WT32.

**Comment by khoih-prog** [comment]
It's much easier to use **ESP32** code, and convert to use for **WT32_ETH01**. I haven't done this, but even don't expect any issue.

You can either use the [**esp32_https_server** library](https://github.com/fhessel/esp32_https_server) or [**ESP32: HTTPS web server** tutorial
](https://techtutorialsx.com/2019/04/07/esp32-https-web-server/) as a start for the conversion.

Just follow and compare the codes of **ESP32** and **WT32_ETH01**, then you'll know what to do.

For example ESP32's [**HelloServer**](https://github.com/espressif/arduino-esp32/tree/master/libraries/WebServer/examples/HelloServer) and WT32_ETH01's [**HelloServer**](https://github.com/khoih-prog/WebServer_WT32_ETH01/tree/main/examples/HelloServer)

Good Luck,

**Comment by khoih-prog** [comment]
Hi @PyCoder040 

I just created a fork of [**esp32_https_server**](https://github.com/fhessel/esp32_https_server) at [**KH forked ESP32_HTTPS_Server**](https://github.com/khoih-prog/esp32_https_server) to fix the compile issue and add examples to support WT32_ETH01

Tested OK so far. Please give it a try and inform if there is any error.

**Reply by PyCoder040** [reply]
This worked perfectly for me!  Thank you!  So far, I have tested the Self-Signed-Certificate and Static-Page examples.

**Comment by khoih-prog** [comment]
Hi @PyCoder040 

Please test the new [**ESP32_HTTPS_Server** v1.1.1](https://github.com/khoih-prog/ESP32_HTTPS_Server/releases/tag/v1.1.1), with the following `changelog`

I'm thinking about create a new `HTTPS_Server_Generic` library to support many more new and powerful boards, such as Portenta_H7, Teensy 4.x, Raspberry Pi Pico_W, etc.

---

1. Merge some upstream PRs

- Handling of errors - like unstable network - coming via SSL https://github.com/fhessel/esp32_https_server/pull/89
- WIP: Prevent crash on WebSocket request to non-WebSocket node. https://github.com/fhessel/esp32_https_server/pull/106
- Fix infinite loop when the buffer ends with \r https://github.com/fhessel/esp32_https_server/pull/123
- Fixed memory leak in the Websocket example https://github.com/fhessel/esp32_https_server/pull/157

2. Update examples and `README.md`
